packages/datastore/src/client.ts

```diff
@@ -229,18 +213,21 @@ export class CollaborationClient extends WSConnection<
   private _handleTransactions(
     transactions: ReadonlyArray<Collaboration.SerialTransaction>
   ) {
-    if (this.handler !== null) {
-      for (let t of transactions) {
-        if (t.serial !== this._serverSerial + 1) {
-          // Out of order serials!
-          // Something has gone wrong somewhere.
-          // TODO: Trigger recovery?
-          throw new Error('Critical! Out of order transactions in datastore.');
-        }
-        this._serverSerial = t.serial;
-        const message = new CollaborationClient.RemoteTransactionMessage(t);
-        MessageLoop.postMessage(this.handler, message);
+    if (!this.handler) {
+      return;
+    }
+    for (let t of transactions) {
+      if (t.serial !== this._serverSerial + 1) {
+        // Out of order serials!
+        // Something has gone wrong somewhere.
+        // TODO: Trigger recovery?
+        throw new Error('Critical! Out of order transactions in datastore.');
```

Comment:
@vidartf Did we ever decide whether this was actually true? It's my impression that things can be out-of-order at the transaction level as well as the patch level.

Reply:
Yeah, this was based on an incorrect assumption about which algorithm was being used (triggered by missing graveyard implementation). The entire serial setup needs a pass. It is probably still needed for other things, but this exception will likely go away.

I can look at it later hopefully.

Reply:
Yeah I didn't fully grok the serial interface. I tried to leave mainly as is for now.